### PR TITLE
signalfxexporter: Update exclude_metrics option to use dpfilters.MetricFilter

### PR DESCRIPTION
**Pull request by asuresh4**
Depends on #1950 

**Description:** Update `exclude_metrics` config option to use `dpfilters.MetricFilter`. This is breaking change to exclude_metrics config option. Now, it takes slice of metric filters instead of just metric names (slice of strings).

**Testing:** Updated tests.

**Documentation:** Updated README.

Not included in this PR but subsequent changes I'm looking to make
1.  Add hard coded list of metrics to be excluded to omit non-default metrics
2.  Add an include_metrics options that will serve as an override to the exclude list. This will help to include metrics that are excluded by default.

### CODE REVIEW

**Comment by asuresh4** [comment]
cc @jrcamp

File: exporter/signalfxexporter/translation/dpfilters/datapoint.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+import (
+	"errors"
+
+	sfxpb "github.com/signalfx/com_signalfx_metrics_protobuf/model"
+)
+
+type dataPointFilter struct {
+	metricFilter     *stringFilter
+	dimensionsFilter *dimensionsFilter
```
Comment (rmfitzpatrick) [inline exporter/signalfxexporter/translation/dpfilters/datapoint.go:25]: Are these pointers for performance reasons?

File: exporter/signalfxexporter/translation/dpfilters/dimensions.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	sfxpb "github.com/signalfx/com_signalfx_metrics_protobuf/model"
+)
+
+type dimensionsFilter struct {
+	filterMap map[string]*stringFilter
+}
+
+// newDimensionsFilter returns a filter that matches against a
+// sfxpb.Dimension slice. The filter will return true, if there's
+// at least one matching dimension in the provided slice or if the
```
Comment (rmfitzpatrick) [inline exporter/signalfxexporter/translation/dpfilters/dimensions.go:29]: As implemented doesn't dimensionsFilter.Matches return true iff all dimensions match and the slice isn't empty?

Comment (asuresh4) [reply exporter/signalfxexporter/translation/dpfilters/dimensions.go]: I believe this branch had an earlier version of the code. Rebased to pick latest commits from PR that was already merged. `dimensionsFilter.Matches` would return `true` if at least one dimension matches and if it's an empty slice, it should return `false` (corrected the comment).

File: exporter/signalfxexporter/translation/dpfilters/datapoint.go (hunk 1 above)
Comment (asuresh4) [reply exporter/signalfxexporter/translation/dpfilters/datapoint.go]: Not really. I guess I just started out with them. Would you advocate for one vs the other?

File: exporter/signalfxexporter/translation/dpfilters/dimensions.go (hunk 2 above)
Comment (asuresh4) [reply exporter/signalfxexporter/translation/dpfilters/dimensions.go]: I stand corrected. The new version also matches on all dimensions. Will update the comment to reflect that. Thanks for catching!

Comment (asuresh4) [reply exporter/signalfxexporter/translation/dpfilters/dimensions.go]: I stand corrected. I've updated the comment to rectify. Thanks for catching!

**Review by jrcamp — COMMENTED** [review]
lgtm

File: exporter/signalfxexporter/testdata/config.yaml [hunk 3]
```diff
@@ -29,6 +29,17 @@ exporters:
...
         k8s.cluster.name: kubernetes_cluster
+    exclude_metrics:
+      - metric_name: metric1
+      - metric_names: [metric2, metric3]
+      - metric_name: metric4
+        dimensions:
+          dimension_key: dimension_val
+      - metric_name: metric5
+        dimensions:
+          dimension_key: [dimension_val1, dimension_val2]
```
[the remark this answers is not in the document]
Comment (asuresh4) [reply exporter/signalfxexporter/testdata/config.yaml:40]: That's a good point. Added a few examples.

**Review by jrcamp — APPROVED** [review]

**Pull request merged**